src/rust/aggregator/src/data/leaderboards.rs

```diff
@@ -0,0 +1,293 @@
+use anyhow::anyhow;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, Duration, Utc};
+use sqlx::{PgConnection, PgPool, Postgres, Transaction};
+
+use super::{Data, DataAggregationError, DataAggregationResult};
+
+pub struct Leaderboards {
+    pool: PgPool,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+impl Leaderboards {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+struct Competition {
+    id: i32,
+    start: DateTime<Utc>,
+    end: DateTime<Utc>,
+    prize: i64,
```

Comment:
@alnoki How are we going to disburse the prize without contact information? Just send the prize money in USDC/USDT to the contending wallet that won?

Reply:
This issue does not concern implementation details